File: src/main/scala/mmlTests/a201504/scala/gpt35turbo/TreeFactoryImpl.scala
package mmlTests.a201504.scala.gpt35turbo

import mmlTests.a201504.scala.Tree
import mmlTests.a201504.scala.TreeFactory
import mmlTests.a201504.scala.gpt35turbo.TreeImpl

class TreeFactoryImpl[X] extends TreeFactory[X] {
  override def emptyTree(): Tree[X] = new TreeImpl[X](null.asInstanceOf[X], Nil)

  override def consTree(root: X, sons: List[Tree[X]]): Tree[X] = new TreeImpl[X](root, sons)
}

File: src/main/scala/mmlTests/a201504/java/gpt35turbo/TreeFactoryImpl.java
package mmlTests.a201504.java.gpt35turbo;

import mmlTests.a201504.java.Tree;
import mmlTests.a201504.java.TreeFactory;

import java.util.ArrayList;
import java.util.List;

public class TreeFactoryImpl<X> implements TreeFactory<X> {

    @Override
    public Tree<X> emptyTree() {
        return new TreeImpl<X>(null, new ArrayList<Tree<X>>());
    }

    @Override
    public Tree<X> consTree(X root, List<Tree<X>> sons) {
        return new TreeImpl<X>(root, sons);
    }

}